src/blocks/HighlightBlock/Component.tsx

```diff
@@ -0,0 +1,92 @@
+import { Article, Category, HighlightBlock as HighlightBlockProps } from '@/payload-types'
+import React from 'react'
+import { getPayload } from 'payload'
+import config from '@payload-config'
+import { HighlightedElement, HighlightElementsComponent } from './elements/Component'
+
+type Props = HighlightBlockProps
+
+const getProductElements = async (productArray: any[]): Promise<HighlightedElement[]> => {
+  const ids = productArray
+    ?.map((product) => {
+      if (product === null || product === undefined) return null
+      return typeof product === 'number' ? product : product.product
+    })
+    .filter((el) => typeof el === 'number')
```

Comment:
I didn't see any products. Quick refactor seems to work:
```suggestion
  const ids = productArray
    ?.map((product) => {
      if (!product) return null
      return typeof product.product === 'number' ? product.product : (product.product?.id ?? null)
    })
    .filter((el): el is number => el !== null)
```